crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -432,20 +432,34 @@ impl EventHandler {
                     "Intercepted HTLC"
                 );
 
+                let fake_channel_payments = self.fake_channel_payments.lock().unwrap();
+
+                let target_node_id = match fake_channel_payments.get(&requested_next_hop_scid) {
+                    None => {
+                        tracing::warn!(fake_scid = requested_next_hop_scid, "Could not forward the intercepted HTLC because we didn't have a node registered with said fake scid");
+
+                        if let Err(err) = self.channel_manager.fail_intercepted_htlc(intercept_id) {
+                            tracing::error!("Could not fail intercepted htlc {err:?}")
+                        }
+
+                        return Ok(());
+                    }
+                    Some(target_node_id) => target_node_id,
+                };
+
                 // if we have already a channel with them, we try to forward the payment.
-                // TODO: here we would need to increase the channel size if the channel is too small
-                if let Some(channel) =
-                    self.channel_manager
-                        .list_channels()
-                        .iter()
-                        .find(|channel_details| {
-                            if let Some(scid) = channel_details.short_channel_id {
-                                scid == requested_next_hop_scid
-                            } else {
-                                false
-                            }
-                        })
+                if let Some(channel) = self
+                    .channel_manager
+                    .list_channels()
+                    .iter()
+                    .find(|channel_details| channel_details.counterparty.node_id == *target_node_id)
                 {
+                    if channel.outbound_capacity_msat < expected_outbound_amount_msat {
+                        // TODO: here we would need to increase the channel size if the channel is
+                        // too small
+                        tracing::warn!("Cannot forward payment on channel, due to insufficient outbound capacity.")
+                    }
```

Comment:
🤔 Looking back on this, I still don't like checking the pre-condition here. This is the job of the `forward_intercepted_htlc` API we call a few lines below.

I think this just leads to:

- Redundant error/warning messages if we emit this log and subsequently fail below.
- Confusing logs, where we log a warning but still succeed below for some unforeseen reason.

Even keeping the TODO is debatable for me, because it's not the only way that we can solve the problem of not having enough liquidity to route a payment to an app. For example, we could alternatively fail during invoice creation, since we know the inbound liquidity of our channel with the coordinator, and show the user an option to explicitly increase the liquidity on the UI.

Given all this, I'd be happy if this bit of code was removed.

Reply:
I like the idea with moving the check to the invoice. However, this approach is also not 100% bullet proof as the user might create multiple invoices.

Reply:
I removed the if condition and added a comment to the `forward_intercepted_htlc` function call.

Reply:
> I like the idea with moving the check to the invoice. However, this approach is also not 100% bullet proof as the user might create multiple invoices.

Yeah, you're right! I was primarily trying to argue that the TODO is a bit too opinionated for me at this stage.